### PR TITLE
Add spans flyout and data grid to view/filter attributes, switch filters to sessionstorage

### PR DESCRIPTION
**Pull request by joshuali925**
*Issue #, if available:*
- https://github.com/opensearch-project/trace-analytics/issues/15
- #36 

*Description of changes:*
- use sessionstorage instead of localstorage, filters are no longer preserved for different sessions
- Add flyout to display span information with filtering
- Add data grid to display spans in trace and service view
- Will add more IT and UT tests later

image

image

 By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by mengweieric — APPROVED** [review]

File: public/components/traces/span_detail_flyout.tsx [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            </EuiFlexItem>
+            <EuiFlexItem>{span.spanId}</EuiFlexItem>
+          </EuiFlexGroup>
+        ) : (
+          '-'
+        )
+      ),
+      getListItem(
+        'parentSpanId',
+        'Parent Span ID',
```
Comment (davidcui-amzn) [inline public/components/traces/span_detail_flyout.tsx:94]: np: `Parent span ID` vs `Parent Span ID`?

File: public/components/traces/span_detail_flyout.tsx [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+      </>
+    );
+  };
+
+  return (
+    <>
+      <EuiFlyout onClose={props.closeFlyout} size="s">
+        <EuiFlyoutHeader hasBorder>
+          <EuiTitle>
+            <h2>Span Detail</h2>
```
Comment (davidcui-amzn) [inline public/components/traces/span_detail_flyout.tsx:203]: Same here, `Span detail` vs `Span Detail`?

File: public/requests/traces_request_handler.ts [hunk 3]
```diff
@@ -191,7 +208,7 @@ const hitsToSpanDetailData = async (hits, colorMap) => {
     const duration = _.round(nanoToMilliSec(hit._source.durationInNanos), 2);
     const serviceName = _.get(hit, ['_source', 'serviceName']);
     const name = _.get(hit, '_source.name');
-    const error = hit._source['status.code'] ? 'Error' : '';
+    const error = hit._source['status.code'] === 2 ? ' \u26a0 Error' : '';
```
Comment (davidcui-amzn) [inline public/requests/traces_request_handler.ts:211]: what does the 2 represent in this expression?

**Review by davidcui-amzn — APPROVED** [review]
LGTM! Just a few minor comments and 1 question

File: public/components/traces/span_detail_flyout.tsx (hunk 1 above)
Comment (joshuali925) [reply public/components/traces/span_detail_flyout.tsx:94]: good catch, updated. thanks

File: public/requests/traces_request_handler.ts (hunk 3 above)
Comment (joshuali925) [reply public/requests/traces_request_handler.ts:211]: Currently data-prepper uses `status.code` = 0 to indicate unset, 1 is success and 2 is error. We only show error on UI if it's 2

**Pull request merged**